docs: readme

danielorihuela opened the pull request:
## Description
Improve the README. Close #16.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

danielorihuela commented:
@rasswanth-s and I have started working on a better readme. We encourage people to give feedback and help us improve it.

jmaunon commented:
> @rasswanth-s and I have started working on a better readme. We encourage people to give feedback and help us improve it.
> 
> **Note**: I started it in markdown mode to make my life easier. However, @jmaunon recomended to use rst to later on use it in the Sphinx documentation.

Hi @danielorihuela . If you want to use markdown is fine (it has no so many options as .rst but I agree that for most of the situations is enough). The only point is that a plugint would be needed (see for example: https://pypi.org/project/m2r2/)

gmuraru commented on README.md:
```diff
@@ -1,16 +1,37 @@
+2. Search for an issue in which you would like to work. Issues for newcomers are labeled with `good first issue`.
+3. Create a PR solving the issue.
 
-This library should not be used in a production environment because it is still a prototype.
```
Could you also leave this?

gmuraru commented:
@danielorihuela is this still WIP?

danielorihuela commented:
Yes, now that I ended the other. I am gonna try and improve this.

danielorihuela commented:
@jmaunon I added some sections more. I do not know if would make sense to add even more sections.
I also added the rest version of the markdown readme. I do no have clear if I should let both readme or just keep the rest version.

jmaunon commented:
@danielorihuela . Just keep one, it you already have in .rst I would keep this one because is easier to integrate with sphinx

Syzygianinfern0 commented on README.rst:
```diff
@@ -0,0 +1,68 @@
+    <br>
+    <a href="http://duet.openmined.org/"><img src="https://github.com/OpenMined/design-assets/raw/master/logos/OM/mark-primary-trans.png" alt="SyMPC" width="200"></a>
+    <br>
+    SyMPC
+    <br>
+    </h1>
+    <h3 align="center">
+    <br>
+    A library that extends PySyft with SMPC support
+    <br>
+    </h3>
```
```suggestion
    SyMPC
    </h1>
    <h4 align="center">
    A library that extends PySyft with SMPC support
    <br>
    <br>
    </h4>
```

| Before | After |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/31875325/122169064-a09f1e00-ce9a-11eb-8954-291f81c47f14.png) | ![image](https://user-images.githubusercontent.com/31875325/122168910-73eb0680-ce9a-11eb-8852-018b1d3d1b18.png) |

Syzygianinfern0 commented on README.rst:
```diff
@@ -0,0 +1,68 @@
+Installation
+############
+
+SyMPC is a companion library for PySyft. Therefore, we will need to install pysyft among other dependencies. We recommend using a virtual environment like conda.
```
```suggestion
SyMPC is a companion library for PySyft. Therefore, we will need to install PySyft among other dependencies. We recommend using a virtual environment like `conda`.
```

Syzygianinfern0 commented on README.rst:
```diff
@@ -0,0 +1,68 @@
+    <a href="https://codecov.io/gh/OpenMined/SyMPC"><img src="https://codecov.io/gh/OpenMined/SyMPC/branch/main/graph/badge.svg?token=TS2rZyJRlo" /></a>
+
+
+SyMPC **/ˈsɪmpəθi/** is a library which extends `PySyft <https://github.com/OpenMined/PySyft>`_ ≥0.3 with SMPC support. It allows to compute over encrypted data, to evaluate and to train neural networks.
```
```suggestion
SyMPC **/ˈsɪmpəθi/** is a library which extends `PySyft <https://github.com/OpenMined/PySyft>`_ ≥0.3 with SMPC support. It allows computing over encrypted data, and to train and evaluate neural networks.
```

danielorihuela replied:
Thanks for the feedback @Syzygianinfern0 !

Syzygianinfern0 reviewed (changes requested):
Minor changes to look, wording

danielorihuela commented:
> Sorry for the late review @danielorihuela.
> This looks great. One small addition you could make is to add instructions to run with docker.
> 
> Building
> `docker build -t sympc -f docker-images/Dockerfile . `
> 
> Running
> `docker run -i -t sympc`

Do not need to apologize! Check the docker section, see if you like it.

jmaunon commented:
@kamathhrishi @danielorihuela , I have reviewed the docker section and it seems confusing. If you run those commands...later what? I have to enter into the docker container and... run the examples? (I do not think so because a Jupiter server is not running); start to develop? . It could be useful to include just a sentence with the next step you would expect from the person who reads that section

One more thing: `docker` section is actually a subsection of `Installation`, isn't it? (Actually is another way of installing sympc)

danielorihuela commented:
@jmaunon 

> @kamathhrishi @danielorihuela , I have reviewed the docker section and it seems confusing. If you run those commands...later what? I have to enter into the docker container and... run the examples? (I do not think so because a Jupiter server is not running); start to develop? . It could be useful to include just a sentence with the next step you would expect from the person who reads that section
> 
> One more thing: `docker` section is actually a subsection of `Installation`, isn't it? (Actually is another way of installing sympc)

If I am not wrong, the docker image gives us another different isolated environment where we can do whatever we want. Let it be execute jupyter notebooks (if we use the correct options), run the tests, make our own experiments or modify the source code to contribute. I tried to make to reflect this ideas with a new structure following your recommendations @jmaunon. However, I am not sure if they are clearly enough or not. Also, if @kamathhrishi could check that I did not write anything technically wrong, that would be great.

kamathhrishi commented:
@danielorihuela  You are right about the docker description. Just that I think the description could be worded like this:

_"If we cannot or do not want to use a virtual environment, we can use a docker image."_ (Don't use this, because people run docker for various reasons)

```
[keep the install docker part]

You can also run SyMPC using docker by running the following commands. 

.. code:: bash       
 $ docker build -t sympc -f docker-images/Dockerfile .   #builds image named sympc
$ docker run -i -t sympc #runs the container 
```

 Sorry I accidentally edited your comment by mistake.

rasswanth-s commented on README.rst:
```diff
@@ -0,0 +1,92 @@
+    <br>
+    SyMPC
+    </h1>
+    <h4 align="center">
```
```suggestion
    <h2 align="center">
```

rasswanth-s commented on README.rst:
```diff
@@ -0,0 +1,92 @@
+    A library that extends PySyft with SMPC support
+    <br>
+    <br>
+    </h4>
```
```suggestion
    </h2>
```

rasswanth-s reviewed (approved):
Great work :+1:

kamathhrishi commented:
> @kamathhrishi @danielorihuela , I have reviewed the docker section and it seems confusing. If you run those commands...later what? I have to enter into the docker container and... run the examples? (I do not think so because a Jupiter server is not running); start to develop? . It could be useful to include just a sentence with the next step you would expect from the person who reads that section
> 
> One more thing: `docker` section is actually a subsection of `Installation`, isn't it? (Actually is another way of installing sympc)

@jmaunon Sorry I didn't see your comment. Yeah, run examples or write and run code. Just that it helps have an environment that would surely work irrespective of platform. Its kind of under installation section. 

@danielorihuela You can place it under installation section with the description I mentioned above.

danielorihuela commented:
> > @kamathhrishi @danielorihuela , I have reviewed the docker section and it seems confusing. If you run those commands...later what? I have to enter into the docker container and... run the examples? (I do not think so because a Jupiter server is not running); start to develop? . It could be useful to include just a sentence with the next step you would expect from the person who reads that section
> > One more thing: `docker` section is actually a subsection of `Installation`, isn't it? (Actually is another way of installing sympc)
> 
> @jmaunon Sorry I didn't see your comment. Yeah, run examples or write and run code. Just that it helps have an environment that would surely work irrespective of platform. Its kind of under installation section.
> 
> @danielorihuela You can place it under installation section with the description I mentioned above.

I have made a change. But I do not know if that's what you wanted @kamathhrishi.

kamathhrishi commented:
@danielorihuela Looks good. Thanks for this PR :)

kamathhrishi commented:
@gmuraru Want to have a look before merging?
